pkg/migration/generate.go

```diff
@@ -830,6 +835,45 @@ func (h *Handler) addMonitoringSpec(cluster *corev1.StorageCluster, ds *appsv1.D
 	return nil
 }
 
+func (h *Handler) addCustomAnnotations(cluster *corev1.StorageCluster, ds *appsv1.DaemonSet) error {
+	podList := &v1.PodList{}
+	err := h.client.List(
+		context.TODO(),
+		podList,
+		&client.ListOptions{
+			Namespace:     ds.Namespace,
+			LabelSelector: labels.SelectorFromSet(map[string]string{"name": "portworx"}),
+		},
+	)
+	if err != nil {
+		return err
+	}
+
+	customAnnotations := make(map[string]string)
+	for _, pod := range podList.Items {
+		owner := metav1.GetControllerOf(&pod)
+		if owner != nil && owner.UID == ds.UID && pod.Annotations != nil {
+			for k, v := range pod.Annotations {
```

Comment:
there is no need to get pod, annotation is in daemonset.

Reply:
I thought user could just annotate the pod manually. so do we look at the daemonset only? or look at both the pods and daemonset? @haibinxie

Reply:
the user wont have access to pod when first deploying it. if users change the pod later directly, then when they upgrade portworx (px version or other stuff), the annotation will be changed back during daemonset rolling upgrade. In any case, I think it's correct to refer to pod as source of truth. so let's keep it unchanged.

Reply:
I agree with Michael about just checking daemonset. If customers are adding annotation directly to pods, then it is not our responsibility to copy them as they are anyway ephemeral. When the DS is updated, the annotations are going to go away. We should only pick up the persisted annotations from the DS.